webrender/src/profiler.rs

```diff
@@ -475,20 +566,29 @@ impl Profiler {
         self.draw_counters(&[
             &backend_profile.total_time,
             &renderer_timers.cpu_time,
-            &renderer_timers.gpu_time_paint,
-            &renderer_timers.gpu_time_composite,
-            &renderer_timers.gpu_time_total,
+            &renderer_timers.gpu_time,
         ], debug_renderer, false);
 
+        let mut gpu_time = 0;
+        let gpu_samples = mem::replace(&mut renderer_timers.gpu_samples, Vec::new());
+        for sample in &gpu_samples {
+            gpu_time += sample.time_ns;
+        }
+
         self.backend_time.push(backend_profile.total_time.nanoseconds);
         self.compositor_time.push(renderer_timers.cpu_time.nanoseconds);
-        self.gpu_time.push(renderer_timers.gpu_time_total.nanoseconds);
+        self.gpu_time.push(gpu_time);
+        self.gpu_frames.push(gpu_time, gpu_samples);
 
         let rect = self.backend_time.draw_graph(self.x_left, self.y_left, "CPU (backend)", debug_renderer);
         self.y_left += rect.size.height + 10.0;
         let rect = self.compositor_time.draw_graph(self.x_left, self.y_left, "CPU (compositor)", debug_renderer);
         self.y_left += rect.size.height + 10.0;
         let rect = self.gpu_time.draw_graph(self.x_left, self.y_left, "GPU", debug_renderer);
         self.y_left += rect.size.height + 10.0;
+        let rect = self.gpu_frames.draw(self.x_left,
+                                        self.y_left,
+                                        debug_renderer);
+        self.y_left += rect.size.height + 10.0;
```

Comment:
While you're here, could you move 10.0 to being a constant?

We'll appreciate this if we port to different sized screens :)